src/main/java/org/logstash/dissect/Dissector.java

```diff
@@ -11,31 +15,30 @@
 import java.util.regex.Pattern;
 
 public class Dissector {
-    private final Pattern pattern = Pattern.compile("(.*?)%\\{(.*?)\\}");
+    private final Pattern delimiterFieldPattern = Pattern.compile("(.*?)%\\{(.*?)\\}");
     private final List<Field> fields = new ArrayList<>();
     private final List<Field> saveableFields = new ArrayList<>();
-    private final Map<Field, ValueRef> fieldValueRefMap = new HashMap<>();
     private int initialOffset = 0;
     private Field lastField;
 
     public Dissector(String mapping) {
-        if (mapping.isEmpty()) return;
+        if (mapping.isEmpty()) {
+            throw new IllegalArgumentException("The mapping string cannot be empty");
+        }
 
-        fieldValueRefMap.put(NormalField.getMissing(), new ValueRef());
-        Matcher m = pattern.matcher(mapping);
+        Matcher m = delimiterFieldPattern.matcher(mapping);
 
         for (int i = 0; m.find(); i++) {
-            Delim delim = DelimFactory.create(m.group(1));
+            Delimiter delimiter = DelimiterFactory.create(m.group(1));
             Field field = FieldFactory.create(m.group(2));
             if (fields.size() == 0) {
```

Comment:
A lot of people use `x.size() == 0` when in fact they want to check `x.isEmpty()`. :) To me the latter is clearer and it may also be better from a performance perspective (not that it matters in general but I just wanted to point it out).